packages/core-utils/src/app/contract-deployment.ts

```diff
@@ -108,3 +117,28 @@ export const deploy = async (
 
   return deployContractFunction(wallet)
 }
+
+/**
+ * Returns an address's first deployed contract or `null` if that address hasn't
+ * created a contract yet. This is useful when creating a contract when a service
+ * is initialized and using that contract on subsequent runs.
+ * @param addresss The address that is being deployed from
+ * @returns contractAddress The address of the first deployed contract or `null` if one hasn't been deployed yet 
+ */
+export const getFirstDeployedContractAddress = async (
+  provider: Provider,
+  address: string,
+): Promise<string | null> => {
+  const nonce = 0
+
+  const contractAddress = getAddress(add0x(keccak256(RLP.encode([
+    getAddress(address),
+    stripZeros(hexlify(nonce))
+  ]))).substring(26));
+
+  if(await provider.getCode(contractAddress) === "0x") {
+    return null
```

Comment:
I think we've been returning `undefined` in these cases. If you have a strong opinion about `null`, let me know, and we can adjust. It's nice that `undefined` is the default for vars so using it makes it so we don't have to check `if (x === null || x === undefined)`.

Reply:
Yeah I think semantically `null` is suppose to mean "nothing" and "undefined" is supposed to mean "has not been defined yet". Ideally JS/TS would just have one empty type. Regardless I'm on board as long as we're consistent and I agree only having to check for one value is simpler 👍-> [Return `undefined` instead of `null` if a contract hasn't been deployed](https://github.com/ethereum-optimism/optimism-monorepo/pull/48/commits/9746b971196fd1b48626c5d4dcbfbb893ece5805)